src/transformers/generation/utils.py

```diff
@@ -2241,6 +2241,10 @@ def greedy_search(
             model_kwargs = self._update_model_kwargs_for_generation(
                 outputs, model_kwargs, is_encoder_decoder=self.config.is_encoder_decoder
             )
+            #add next_tokens to inputs_embeds if using embeds to generate
+            if model_kwargs.get("decoder_inputs_embeds") is not None:
+                next_tokens_embed = self.decoder.get_input_embeddings()(next_tokens) * self.decoder.model.decoder.embed_scale
+                model_kwargs["decoder_inputs_embeds"] = torch.cat([model_kwargs["decoder_inputs_embeds"], next_tokens_embed.unsqueeze(1)], dim=-2)
```

Comment:
I will not accept these changes to the body of generate :)

There are multiple reasons for this decision:
a) A similar proposal is being tracked [here](https://github.com/huggingface/transformers/issues/6535#issuecomment-1365401415). It will only be added for consideration after it raises sufficient interest, as described in the link;
b) We want to avoid adding more logic to `generate` itself unless it is a widely requested feature or it can be added as part of the model itself (e.g. in `prepare_inputs_for_generation`) / in a self-contained class (like the `LogitsProcessors`);